features/lorawan/system/LoRaWANTimer.h

```diff
@@ -0,0 +1,115 @@
+/**
+ / _____)             _              | |
+( (____  _____ ____ _| |_ _____  ____| |__
+ \____ \| ___ |    (_   _) ___ |/ ___)  _ \
+ _____) ) ____| | | || |_| ____( (___| | | |
+(______/|_____)_|_|_| \__)_____)\____)_| |_|
+    (C)2013 Semtech
+
+Description: Timer objects and scheduling management
+
+License: Revised BSD License, see LICENSE.TXT file include in the project
+
+Maintainer: Miguel Luis and Gregory Cristian
+
+
+Copyright (c) 2017, Arm Limited and affiliates.
+
+SPDX-License-Identifier: BSD-3-Clause
+*/
+
+#ifndef MBED_LORAWAN_SYS_TIMER_H__
+#define MBED_LORAWAN_SYS_TIMER_H__
+#include "drivers/Timer.h"
+#include "drivers/Ticker.h"
+
+using namespace mbed;
+
+/*!
+ * \brief Timer object description
+ */
+typedef struct TimerEvent_s
+{
+    uint32_t value;
+    void ( *Callback )( void );
+    Ticker Timer;
+}TimerEvent_t;
+
+/*!
+ * \brief Timer time variable definition
+ */
+#ifndef TimerTime_t
+typedef uint32_t TimerTime_t;
+#endif
+
+/*!
+ * \brief Initializes the timer used to get the current time.
+ *
+ * \remark The current time corresponds to the time since system startup.
+ */
+void TimerTimeCounterInit( void );
+
+/*!
+ * \brief Initializes the timer object.
+ *
+ * \remark The TimerSetValue function must be called before starting the timer.
+ *         This function initializes the timestamp and reloads the value at 0.
+ *
+ * \param [in] obj          The structure containing the timer object parameters.
+ * \param [in] callback     The function callback called at the end of the timeout.
+ */
+void TimerInit( TimerEvent_t *obj, void ( *callback )( void ) );
+
+/*!
+ * \brief Starts and adds the timer object to the list of timer events.
+ *
+ * \param [in] obj The structure containing the timer object parameters.
+ */
+void TimerStart( TimerEvent_t *obj );
+
+/*!
+ * \brief Stops and removes the timer object from the list of timer events.
+ *
+ * \param [in] obj The structure containing the timer object parameters.
+ */
+void TimerStop( TimerEvent_t *obj );
+
+/*!
+ * \brief Resets the timer object.
+ *
+ * \param [in] obj The structure containing the timer object parameters.
+ */
+void TimerReset( TimerEvent_t *obj );
+
+/*!
+ * \brief Set a new timeout value.
+ *
+ * \param [in] obj   The structure containing the timer object parameters.
+ * \param [in] value The new timeout value.
+ */
+void TimerSetValue( TimerEvent_t *obj, uint32_t value );
+
+/*!
+ * \brief Read the current time.
+ *
+ * \retval time The current time.
+ */
+TimerTime_t TimerGetCurrentTime( void );
```

Comment:
If this is not C++ class, then these should be written as `timer_time_t timer_get_current_time(void);`. Applies to the all API in this header file

Reply:
Same problem here. These APIs are being used in LoRaMac,cpp and we don't want to change the code there unless absolutely necessary.